queries/leo/highlights.scm

```diff
@@ -0,0 +1,262 @@
+(identifier) @variable 
+
+[
+ "assert"
+ "assert_eq"
+ "assert_neq"
+ "block"
+ "console"
+ "const"
+ "constant"
+ "finalize"
+ "in"
+ "let"
+ "mapping"
+ "private"
+ "program"
+ "public"
+ "record"
+ "self"
+ "struct"
+] @keyword
+
+[
+ "transition"
+ "function"
+ "inline"
+] @keyword.function
+
+"import" @include
+
+"return" @keyword.return
+
+"for" @repeat
+
+[ 
+  "if"
+  "then"
+  "else"
+] @conditional
+
+[
+ (ternary_if)
+ (ternary_else)
+] @conditional.ternary
+
+
+[ ";" "," "::"] @punctuation.delimiter
+
+[
+"!"
+
+"&&"
+"||"
+
+"=="
+"!="
+
+"<"
+"<="
+">"
+">="
+
+"&"
+"|"
+"^"
+
+"<<"
+">>"
+
+"+"
+"-"
+"*"
+"/"
+"%"
+"**"
+
+"="
+
+"+="
+"-="
+"*="
+"/="
+"%="
+"**="
+
+"<<="
+">>="
+
+"&="
+"|="
+"^="
+
+"&&="
+"||="
+
+] @operator
+
+(comment) @comment
+
+(boolean_literal) @boolean
+
+(constant_declaration 
+  (identifier) @constant
+)
+
+;program name if with right extension "aleo" -> okay
+(program_declaration
+  (program_id 
+    name: (identifier) 
+    extension: (identifier) @_ext (#eq? @_ext "aleo")) @string
+)
+
+;program name if with not with extension "aleo" -> error
+(program_declaration
+  (program_id 
+    name: (identifier) 
+    extension: (identifier) @_ext (#not-eq? @_ext "aleo")) @punctuation.delimiter
+)
+
+;record declaration
+(record_declaration (identifier) @field) 
+
+;struct component 
+(struct_component_declaration 
+  (identifier) @field
+) 
+
+(type) @type
+
+(associated_constant) @constant
+
+[
+ (self_caller)
+ (block_height)
+] @constant.builtin
+
+(transition_declaration
+  .
+  name: (identifier) @function.bultin
+  .
+  (function_parameters)*
+  .
+  (return_arrow) @keyword.return
+  .
+  (type)
+  .
+  (block)
+  .
+)
```

Comment:
This simplifies the expressions and makes `(return_arrow)` consistent with other languages.
```suggestion
(return_arrow) @punctuation.delimiter

(transition_declaration
  . name: (identifier) @function.bultin)
```